src/main/java/core/basesyntax/factory/Factory.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax.factory;
+
+import core.basesyntax.dao.BetDao;
+import core.basesyntax.dao.BetDaoImpl;
+import core.basesyntax.dao.CarDao;
+import core.basesyntax.dao.CarDaoImpl;
+import core.basesyntax.exception.NoDaoAnnotationException;
+import core.basesyntax.lib.Dao;
+
+public class Factory {
+    private static BetDao betDao;
+    private static CarDao carDao;
+
+    public static BetDao getBetDao() {
+        if (betDao == null) {
+            if (BetDaoImpl.class.getAnnotation(Dao.class) == null) {
+                throw new NoDaoAnnotationException("BetDaoImpl class has no @Dao annotation");
+            }
+            betDao = new BetDaoImpl();
+        }
+        return betDao;
```

Comment:
I think will be better let Factory just return a instance of class and all logic with reflection put in Injector class.